spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2712 @@
+
+URI indicating the process host’s identity. Ex: GitLab Runner ID
+
+`monitoredProcess.type` _string (TypeURI)_, _required_
+
+URI indicating the type of process performed. Ex: when monitoring a build, this
+field records the build type.
+
+`monitoredProcess.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`monitorLog` _object_, _required_
+
+Record of events that were traced by the monitor. One of `process`, `network`,
+and `fileAccess` must be present.
+
+`monitorLog.process` _list_, _optional_
+
+`monitorLog.network` _list_, _optional_
+
+`monitorLog.fileAccess` _list_, _optional_
+
+Record of files accessed during the build process. A complete list of
+_materials_ can be derived from this information. Each entry in this list is
+expected to record the path of the file and one or more digests of the file.
+This field is a list rather than a key value map because a single file may be
+used multiple times during the build process. Further, some files that are
+accessed may _change_ during the build process, and so, different entries may
+have different digests.
+
+While this predicate can be used to log file accesses, the actual technique used
+to capture the file access event has some implications. If a synchronous
+monitor, for example one that uses `ptrace` to trace the file access system
+calls, is used, then the build process can be paused while the file's digest is
+calculated and stored. However, asynchronous monitors such as those using eBPF
+cannot pause the build process before the file is actually used. Therefore, they
+cannot make as strong guarantees about the digests of the files accessed.
+Verifiers using runtime trace attestations for file accesses must be careful
+about what guarantees they are actually getting based on how the build process
+was monitored.
+
+`metadata` _object_, _optional_
+
+Other properties of the monitoring event.
+
+`metadata.buildStartedOn` _object_, _optional_
+
+`metadata.buildFinishedOn` _object_, _optional_
+
+## Example
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://in-toto.io/attestation/runtime-trace/v0.1",
+  "subject": [
+    {
+      "name": "ttl.sh/testin123",
+      "digest": {
+        "sha256": "def2bdf8cee687d5889d51923d7907c441f1a61958f1e5dfb07f53041c83745f"
+      }
+    }
+  ],
+  "predicate": {
+    "monitor": {
+      "id": "https://tetragon",
+      "configSource": {},
+      "tracePolicy": {
+        "policies": [
+          {
+            "Name": "connect",
+            "Config": ""
+          },
+          {
+            "Name": "sys-read-follow-prefix",
+            "Config": ""
+          }
+        ]
+      }
+    },
+    "monitoredProcess": {
+      "hostID": "https://tekton.dev/chains/v2",
+      "type": "https://tekton.dev/attestations/chains@v2",
+      "event": "run-image-pipelinerun-build-trusted"
+    },
+    "monitorLog": {
+      "process": [
```

Comment:
optional: Would it make any sense to truncate this list?

Reply:
Done.